demos/digit-prediction/digits_prediction.nlogo

```diff
@@ -0,0 +1,648 @@
+extensions [ py ]
+
+globals [
+  coords-x coords-y color-list digit
+  zero one two three four five six seven eight nine
```

Comment:
Rather than storing each probability in it's own variable, you could just store the all the probabilities in a list. I'll show code below.